scripts/webidl-analyze.py

```diff
@@ -0,0 +1,749 @@
+            handle_attribute(records, name, methods, cpp_symbols, member)
+        elif isinstance(member, WebIDL.IDLConst):
+            handle_const(records, name, methods, cpp_symbols, member)
+        elif isinstance(member, WebIDL.IDLMaplikeOrSetlike):
+            handle_maplike_or_setlike_or_iterable(records, name, member)
+        elif isinstance(member, WebIDL.IDLIterable):
+            handle_maplike_or_setlike_or_iterable(records, name, member)
+        elif isinstance(member, WebIDL.IDLAsyncIterable):
+            handle_maplike_or_setlike_or_iterable(records, name, member)
+        else:
+            print(f'WebIDL: Unknown member production: {member.__class__.__name__}',
+                  file=sys.stderr)
+            sys.exit(1)
+
+    emit_structured(records, loc, 'class', pretty, idl_sym,
+                    slots=slots, supers=supers,
+                    methods=methods, fields=fields)
+
+
+def handle_dictionary_field(records, dictionary_name, dictionary_cpp_sym,
+                            fields, target):
+    '''Emit analysis record for IDLArgument in dictionary.'''
+
+    name = target.identifier.name
+    loc = to_loc(target.identifier.location, name)
+    pretty = f'{dictionary_name}.{name}'
+    idl_sym = f'WEBIDL_{dictionary_name}_{name}'
+    cpp_sym = f'F_<{dictionary_cpp_sym}>_{cpp_dictionary_field_name(name)}'
+    js_sym = f'#{name}'
+
+    emit_source(records, loc, 'idl', 'field', pretty, idl_sym)
+    emit_target(records, loc, 'idl', pretty, idl_sym)
+
+    slots = []
+    append_slot(slots, 'attribute', 'cpp', cpp_sym)
+    append_slot(slots, 'attribute', 'js', js_sym)
+
+    emit_structured(records, loc, 'field', pretty, idl_sym,
+                    slots=slots)
+
+    handle_type(records, target.type)
+
+    fields.append({
+        'pretty': pretty,
+        'sym': idl_sym,
+        'props': [],
+    })
+
+
+def handle_dictionary(records, target):
+    '''Emit analysis record for IDLDictionary or IDLPartialDictionary.'''
+
+    name = target.identifier.name
+    loc = to_loc(target.identifier.location, name)
+    pretty = name
+    idl_sym = f'WEBIDL_{name}'
+    cpp_sym = f'T_mozilla::dom::{name}'
+    js_sym = f'#{name}'
+
+    emit_source(records, loc, 'idl', 'dictionary', pretty, idl_sym)
+    emit_target(records, loc, 'idl', pretty, idl_sym)
+
+    slots = []
+    append_slot(slots, 'class', 'cpp', cpp_sym)
+    append_slot(slots, 'class', 'js', js_sym)
+
+    supers = []
+    if hasattr(target, 'parent') and target.parent:
+        handle_super(records, supers, target.parent)
+
+    fields = []
+    for member in target.members:
+        if isinstance(member, WebIDL.IDLArgument):
+            handle_dictionary_field(records, name, cpp_sym,
+                                    fields, member)
+        else:
+            print(f'WebIDL: Unknown member production: {member.__class__.__name__}',
+                  file=sys.stderr)
+            sys.exit(1)
```

Comment:
Meta: Since this is working, should this and the other similar sys.exit() calls just be skipping the top-level thing going on and give a warning that our warning emailing script will pick up instead of exiting from the batch job?

No need to change anything in the PR.